### PR TITLE
Terraform resource for MongoDB Custom Roles

### PR DESCRIPTION
**Pull request by PacoDw**
**Work in progress**. Opening a draft pull request to show progress on this and perhaps occasionally ask for advice.

Adding a new resource for managing Custom MongoDB Roles. This follows my earlier pull request mongodb/go-client-mongodb-atlas#20 adding the necessary functionality to the Go client.

### Resource
- [x] Schema
- [x] Creation
- [x] Reading
- [x] Updating
- [x] Deletion
- [x] Import
- [x] Testing file
- [x] Documentation
### Data source
- [x] Schema
- [x] Reading
- [x] Testing file
- [x] Documentation

### CODE REVIEW

**Comment by PacoDw** [comment]
@mpaluchowski We can continue since this pr :) I was able to recover the previous work 👍

**Comment by PacoDw** [comment]
Hi @mpaluchowski! Good news everything is done, now we need to wait until the review to be able to merge it to the master branch. Thank you so much for your contribution and all your effort, well done!

File: mongodbatlas/resource_mongodbatlas_custom_db_role.go
Comment (marinsalinas) [inline mongodbatlas/resource_mongodbatlas_custom_db_role.go:38]: Nice catch!

File: mongodbatlas/resource_mongodbatlas_custom_db_role.go
Comment (marinsalinas) [inline mongodbatlas/resource_mongodbatlas_custom_db_role.go]: Could you please remove this log?

**Review by marinsalinas — CHANGES_REQUESTED** [review]
LGTM, just remove logs and fix the conflicts, (SDK v0.1.3 is released you are able to use it now)

File: mongodbatlas/resource_mongodbatlas_custom_db_role.go
Comment (PacoDw) [reply mongodbatlas/resource_mongodbatlas_custom_db_role.go]: Nice! thank you so much :) let me remove it 👍

**Comment by themantissa** [comment]
@PacoDw  is this ready for a MDB review?

**Comment by PacoDw** [comment]
yes @themantissa, it is, goes ahead! 👍

File: website/docs/r/custom_db_role.html.markdown
Comment (themantissa) [inline website/docs/r/custom_db_role.html.markdown]: Misspelled IMPORTANT

File: website/docs/r/custom_db_role.html.markdown
Comment (themantissa) [inline website/docs/r/custom_db_role.html.markdown]: Add the link and a bit more detail:
Name of the privilege action. For a complete list of actions available in the Atlas API, see 
[Custom Role Actions] (https://docs.atlas.mongodb.com/reference/api/custom-role-actions).
Note: The privilege actions available to the Custom Roles API resource represent a subset of the privilege actions available in the Atlas Custom Roles UI.

File: website/docs/r/custom_db_role.html.markdown
Comment (themantissa) [inline website/docs/r/custom_db_role.html.markdown]: Misspelling: IMPORTANT

**Review by themantissa — CHANGES_REQUESTED** [review]
A few small edits to the docs but otherwise LGTM!

File: website/docs/r/custom_db_role.html.markdown
Comment (PacoDw) [reply website/docs/r/custom_db_role.html.markdown]: goog catch! thanks :)

**Review by marinsalinas — APPROVED** [review]
LGTM!

**Review by themantissa — APPROVED** [review]
Looks great! Thanks!

**Pull request merged**
